docker-compose.yml

```diff
@@ -1,9 +1,102 @@
-version: '2'
+version: '3'
 services:
+  # MySQL - db
   db:
     image: mysql:5.7
+    hostname: db
     environment:
       - "MYSQL_ROOT_PASSWORD=mysql"
       - "MYSQL_DATABASE=codalab_website"
     ports:
         - 3306:3306
+  
+  # RabbitMQ - rabbit
+  rabbit:
+    image: rabbitmq:3.5.7
+    hostname: rabbit
+    environment:
+      - RABBITMQ_DEFAULT_USER=guest
+      - RABBITMQ_DEFAULT_PASS=guest
+    ports:
+      - "5672:5672" 
+      - "15672:15672"
+
+  # Flower - flower
+  flower:
+    image: totem/celery-flower-docker
+    hostname: flower
+    ports:
+      - "5555:5555"
+    environment:
+      - AMQP_USERNAME=guest
+      - AMQP_PASSWORD=guest
+      - AMQP_HOST=rabbit
+      - AMQP_PORT=5672
+      # - AMQP_ADMIN_USERNAME=admin
+      # - AMQP_ADMIN_PASSWORD=rabbitmq
+      # - AMQP_ADMIN_HOST=rabbit
+      # - AMQP_ADMIN_PORT=15672
+    volumes:
+      - .:/app
+    depends_on:
+      - rabbit
+  
+  # Memcached Container
+  memcached:
+    image: memcached
+    hostname: memcached
+    ports:
+      - "11211:11211"
+    command: "/usr/local/bin/memcached -u memcache -vv"
+  
+  # Django - web
+  web:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    hostname: web
+    command: sh run_web.sh
+    volumes:
+      - .:/app 
+    ports:
+      - "8000:8000"
+    links:
+      - db
+      - rabbit
+      - memcached
+    depends_on:
+      - db
+
+  # Celery Workers
```

Comment:
so long as you dont docker rm a container you should be able to view old logs with docker logs or by accessing the container directly